sdk/go/common/resource/config/object.go

```diff
@@ -69,6 +69,44 @@ func (c object) Secure() bool {
 	}
 }
 
+func decryptMap(ctx context.Context, objectMap map[Key]object, decrypter Decrypter) (map[Key]Plaintext, error) {
+	// Collect all secure values
+	var refs []containerRef
+	var ctChunks [][]string
+	collectCiphertextSecrets(objectMap, &refs, &ctChunks)
+
+	// Decrypt objects in batches
+	offset := 0
+	for _, ctChunk := range ctChunks {
+		if len(ctChunk) == 0 {
+			continue
+		}
+		decryptedChunk, err := decrypter.BatchDecrypt(ctx, ctChunk)
+		if err != nil {
+			return nil, err
+		}
+		// Assign decrypted values back into original structure
+		// We are accepting that a Ciphertext Secret now has a Plaintext value
+		for i, decrypted := range decryptedChunk {
+			refs[offset+i].setObject(newObject(CiphertextSecret{decrypted}))
+		}
```

Comment:
I can see why you've done this this way, but I have to admit that I don't love that we're writing plaintext into `object`s, which per `newObject` are never expected to contain `PlaintextSecret` values.

I don't want to block the PR on this, but I might suggest changing the approach to use a caching decrypter in the future rather than assigning the plaintext back into the `object`s as `CiphertextSecret` values and then using the `NopDecrypter`.